File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/SorpresaSalirCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author pc
 */
import java.util.ArrayList;
class SorpresaSalirCarcel extends Sorpresa{
    
    private MazoSorpresas mazo;
    SorpresaSalirCarcel(MazoSorpresas mazo,String texto){
        super(texto);
        this.mazo=mazo;
    }
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
         if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            Boolean ok=false;
            for(int i=0; i<todos.size();i++)
              if(todos.get(i).tieneSalvoconducto())
                  ok=true;
              if(!ok){
                  todos.get(actual).obtenerSalvoconducto(this);
                  salirDelMazo();
              }
         }
    }
    void salirDelMazo(){
        mazo.inhabilitarCartaEspecial(this);
    }
    @Override
    public String toString() {
        return "Sorpresa Salir Carcel ";
    }
    void usada()
    {
        mazo.habilitarCartaEspecial(this);   
    }
    
}

File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/sorpresa_salir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class SorpresaSalirCarcel < Sorpresa
  def initialize(texto,mazo)
    super(texto)
    @mazo = mazo
  end
  
  def aplicarAJugador(actual,todos)
    if(jugadorCorrecto(actual,todos))
      super(actual,todos)
      estado=false
      contador=0
      while contador < todos.size
        if todos.at(contador).tieneSalvoconducto
          estado=true
        end
        contador+=1
      end
      if(!estado)
        todos.at(actual).obtenerSalvoconducto(self)
        salirDelMazo
      end
    end
  end
  
  def usada 
    @mazo.habilitarCartaEspecial(self)
  end
  
  def salirDelMazo
    @mazo.inhabilitarCartaEspecial(self)
  end
 
  def toString
    a = "Salir Carcel"
    return a
  end
  public_class_method :new
end
end